AzFilesHybrid/AzFilesHybrid.psm1

```diff
@@ -4006,11 +4007,102 @@ function Debug-AzStorageAccountEntraKerbAuth {
+            try {
+                $checksExecuted += 1
+                Write-Verbose "CheckDefaultShareRBACPermission - START"
+
+                $StorageAccountObject = Validate-StorageAccount `
+                    -ResourceGroupName $ResourceGroupName `
+                    -StorageAccountName $StorageAccountName `
+                    -ErrorAction Stop
+ 
+                $DefaultSharePermission = $StorageAccountObject.AzureFilesIdentityBasedAuth.DefaultSharePermission
+                
+                # If DefaultSharePermission is null or 'None'
+                if((!$DefaultSharePermission) -or ($DefaultSharePermission -eq 'None')){
+                    Debug-RBACCheck -StorageAccountName $StorageAccountName -checkResult $checks["CheckDefaultShareRBACPermission"]
+                    #$DefaultSharePermission = "Not Configured. Please visit https://docs.microsoft.com/en-us/azure/storage/files/storage-files-identity-ad-ds-assign-permissions?tabs=azure-portal for more information if needed."
+                }
+                Write-Verbose "CheckDefaultShareRBACPermission: $CheckDefaultShareRBACPermission"
+                Write-Verbose "CheckDefaultShareRBACPermission - SUCCESS"
+                $checks["CheckDefaultShareRBACPermission"].Result = "Passed"
+            } catch {
+                $checks["CheckDefaultShareRBACPermission"].Result = "Failed"
+                $checks["CheckDefaultShareRBACPermission"].Issue = $_
+                Write-Error "CheckDefaultShareRBACPermission - FAILED"
+                Write-Error $_
+            }
+        }
 
         SummaryOfChecks -filterIsPresent $filterIsPresent -checksExecuted $checksExecuted
     }
 }
 
+function Debug-RBACCheck {
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory=$True, Position=0, HelpMessage="Storage account name")]
+        [string]$StorageAccountName,
+        [Parameter(Mandatory=$True, Position=0, HelpMessage="User Principal name")]
+        [string]$UserPrincipalName,
+        [Parameter(Mandatory=$True, Position=1, HelpMessage="Check result object")]
+        [CheckResult]$checkResult
+
+    )
+    process {
+        try {
+            Request-ConnectMsGraph `
+                    -Scopes "User.Read.All", "GroupMember.Read.All" `
+                    -RequiredModules @("Microsoft.Graph.Users", "Microsoft.Graph.Groups", "Microsoft.Graph.Identity.DirectoryManagement")
+                    
+            $userOid = $(Get-MgUser -Filter "UserPrincipalName eq $UserPrincipalName" -Property Id).Id
+
+            $groupIds = Get-MgUserMemberOf -UserId $userOid | Select-Id
+
+            $roleNames = @(
+            "Storage File Data SMB Share Reader",
+            "Storage File Data SMB Share Contributor",
+            "Storage File Data SMB Share Elevated Contributor"
+            )
+            $listOfRoleNames = @{}
+            foreach ($roleName in $roleNames) {
+                $assignments = Get-AzRoleAssignment -RoleDefinitionName $roleName

```

Comment:
We need to also look at the _scope_ of the role assignment. It must be a role assignment that applies to this storage account, and not to other storage accounts. Or alternatively, it should apply to the resource group or subscription, and be inherited.

Reply:
Did some experimentation, something like this should work:

```powershell
$scope = "/subscriptions/$subscription/resourceGroups/$resourceGroup/providers/Microsoft.Storage/storageAccounts/$storageAccount/fileServices/default/fileshares/$fileShare"
$assignments = Get-AzRoleAssignment -RoleDefinitionName $roleName -Scope $scope
```